pkg/client/repo/core.go

```diff
@@ -0,0 +1,59 @@
+package repo
+
+import (
+	"io/ioutil"
+
+	"github.com/bitnami-labs/charts-syncer/pkg/client"
+
+	"github.com/bitnami-labs/charts-syncer/pkg/client/repo/chartmuseum"
+	"github.com/bitnami-labs/charts-syncer/pkg/client/repo/harbor"
+	"github.com/bitnami-labs/charts-syncer/pkg/client/repo/helmclassic"
+	"github.com/bitnami-labs/charts-syncer/pkg/client/repo/local"
+	"github.com/bitnami-labs/charts-syncer/pkg/client/repo/oci"
+
+	"github.com/bitnami-labs/charts-syncer/api"
+	"github.com/bitnami-labs/charts-syncer/internal/cache"
+	"github.com/bitnami-labs/charts-syncer/pkg/client/types"
+	"github.com/juju/errors"
+)
+
+// NewClient returns a Client object
+//
+// The func is exposed as a var to allow tests to temporarily replace its
+// implementation, e.g. to return a fake.
+var NewClient = func(repo *api.Repo, opts ...types.Option) (client.ReadWriter, error) {
```

Comment:
> // The func is exposed as a var to allow tests to temporarily replace its

this does not seem like a good idea, ideally mocking should be done via custom interfaces, in any case I understand that this was this way before so it's out of the scope of this patch to change it.